analysis/grammar_search_analysis.py

```diff
@@ -2,263 +2,188 @@
+                                       run_planning: bool,
+                                       ) -> Dict[str, float]:
+    heuristics = {
+        "prediction_error": _PredictionErrorHeuristic,
+        "hadd_lookahead": _HAddLookaheadHeuristic,
+    }
     utils.flush_cache()
     candidates = {p : 1.0 for p in predicates}
     all_predicates = predicates | initial_predicates
     atom_dataset = utils.create_ground_atom_dataset(dataset, all_predicates)
+    results = {}
     # Compute heuristic scores.
-    for heuristic_cls in [_PredictionErrorHeuristic, _HAddLookaheadHeuristic]:
+    for heuristic_name in heuristic_names:
+        heuristic_cls = heuristics[heuristic_name]
         heuristic = heuristic_cls(initial_predicates, atom_dataset,
                                   candidates)
         heuristic_score = heuristic.evaluate(frozenset(predicates))
-        if len(predicates) >= 10:
-            predicate_str = f"All {len(predicates)} candidates"
-        else:
-            predicate_str = str(predicates)
-        print("\n\n******", env.__class__.__name__, predicate_str,
-              heuristic_cls.__name__, heuristic_score)
+        results[heuristic_name] = heuristic_score
     # Learn NSRTs and plan.
-    utils.flush_cache()
-    approach = create_approach("nsrt_learning", env.simulate, all_predicates,
-                               env.options, env.types, env.action_space)
-    approach.learn_from_offline_dataset(dataset, train_tasks)
-    approach.seed(CFG.seed)
-    _run_testing(env, approach)
+    if run_planning:
+        utils.flush_cache()
+        approach = create_approach("nsrt_learning", env.simulate,
+            all_predicates, env.options, env.types, env.action_space)
+        approach.learn_from_offline_dataset(dataset, train_tasks)
+        approach.seed(CFG.seed)
+        planning_result = _run_testing(env, approach)
+        results.update(planning_result["test"])
+        results.update(planning_result["approach"])
+    return results
+
+
+def _make_proxy_analysis_results(outdir: str) -> None:
+    all_results: DefaultDict[Tuple[str, str], Dict] = defaultdict(dict)
+    for filepath in glob.glob(f"{outdir}/*.result"):
+        with open(filepath, "r", encoding="utf-8") as f:
+            raw_result = f.read()
+        result = float(raw_result)
+        _, filename = os.path.split(filepath[:-len(".result")])
+        env_name, preds, metric = filename.split("__")
+        all_results[(env_name, preds)][metric] = result
+    all_results_table = [{"Env": env_name, "Preds": preds, **metrics}
+                         for (env_name, preds), metrics in all_results.items()]
+    df = pd.DataFrame(all_results_table)
+    print(df)
+    csv_filepath = os.path.join(outdir, "proxy_analysis.csv")
+    df.to_csv(csv_filepath)
+    print(f"Wrote out to {csv_filepath}.")
 
 
 if __name__ == "__main__":
-    # _run_analysis()
-    _run_proxy_analysis()
+    ENV_NAMES = [
```

Comment:
It's a bit strange to have all these constants (all-caps names) defined down here. Two options here:
1) Move all this into a new method called `def _main():` so you can name these constants as regular variables instead, and then just do `if __name__ == "__main__": _main()`
2) Define all these constants at top of file